hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -470,7 +470,7 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	})
 
 	// Validate modifying CPU arch support for platform
-	if (nodePool.Spec.Arch != "amd64") && (nodePool.Spec.Platform.Type != hyperv1.AWSPlatform) {
+	if (nodePool.Spec.Arch != "amd64" && nodePool.Spec.Arch != "ppc64le") && (nodePool.Spec.Platform.Type != hyperv1.AWSPlatform && nodePool.Spec.Platform.Type != hyperv1.AgentPlatform && nodePool.Spec.Platform.Type != hyperv1.PowerVSPlatform) {
```

Comment:
I think you'd want `(nodePool.Spec.Arch != "amd64" || nodePool.Spec.Arch != "ppc64le")`. Actually the whole if statement probably needs reworked. You could skip this validation check on `arch == ppc64le` and the platform is AWS, which is not a supported combo.

I would think the easiest thing to do is convert this if statement to a function that validates the arch based on platform. Then we could easily add a unit test for that to make sure we are using the right platform/arch combos. WDYT?

Reply:
yes Agree @bryan-cox, will refactor this by adding a func like you have described!

Reply:
I have added the func, not sure I have covered all the supported combination.
Please let me know any other combination needs to be updated here?